Referee.cs

```diff
@@ -0,0 +1,1179 @@
+﻿﻿using System;
+using System.Runtime.Serialization;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using UnityEngine;
```

Comment:
the whole point for me was that i'm trying another approach which requires me to run my own simulation of the game in faster iteration cycles with more feedback and i thought i'd share what i've got, also i like the idea of the game a lot